dds/idl/annotations.cpp

```diff
@@ -92,29 +91,29 @@ AST_Expression::AST_ExprValue* get_annotation_member_ev(
 {
   AST_Annotation_Member* member =
     dynamic_cast<AST_Annotation_Member*>((*appl)[member_name]);
-  if (!member) {
-    return 0;
-  }
-
-  AST_Expression* e = member->value();
-  if (!e) {
-    return 0;
+  if (member) {
+    AST_Expression* e = member->value();
+    if (e) {
+      return e->ev();
+    }
   }
 
-  return e->ev();
+  be_util::misc_error_and_abort(std::string("Found null pointer while getting value of member \"") +
+                                member_name + "\" of annotation \"" +
+                                appl->local_name()->get_string() + "\"",
+                                appl);
+  return 0;
 }
 
 bool get_bool_annotation_member_value(AST_Annotation_Appl* appl,
                                       const char* member_name)
 {
   AST_Expression::AST_ExprValue* ev = get_annotation_member_ev(appl, member_name);
   if (!ev) {
-    idl_global->err()->misc_error(
-      (std::string("Found null pointer while getting value of member \"") +
-        member_name + "\" of annotation \"" +
-        appl->local_name()->get_string() + "\"").c_str(),
-      appl);
-    BE_abort();
+    be_util::misc_error_and_abort(std::string("Found null pointer while getting value of member \"") +
+                                  member_name + "\" of annotation \"" +
+                                  appl->local_name()->get_string() + "\"",
+                                  appl);
```

Comment:
I still don't understand why all the callers of `get_annotation_member_ev` have to do the same check before was just done in `get_annotation_member_ev`.

Reply:
@iguessthislldo I'm working on a version of this that passes an `AST_Expression::ExprType` to `get_annotation_member_ev()` so it can be checked.  Does that seem right to you?

Reply:
Yes, how it looks now is fine.